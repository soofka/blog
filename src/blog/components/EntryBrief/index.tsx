import * as React from 'react';
import { Link } from 'react-router-dom';

import { Label } from 'blog/components/Label';

import {
  StyledEntryBriefContainer,
  StyledEntryImage,
  StyledEntryImageWrapper,
} from './styled';

export interface EntryBriefInterface {
  text: string;
  imageUrl?: string;
}

interface EntryBriefPropsInterface extends EntryBriefInterface {
  moreButtonLink?: string;
}

// @todo: fix links
export const EntryBrief = (props: EntryBriefPropsInterface): JSX.Element => {
  const { text, imageUrl, moreButtonLink } = props;

  return (
    <div>
      {imageUrl &&
        <StyledEntryImageWrapper>
          <StyledEntryImage
            src={imageUrl}
          />
        </StyledEntryImageWrapper>
      }
      <StyledEntryBriefContainer dangerouslySetInnerHTML={{ __html: text }} />
      <p>
        {moreButtonLink && <Link to={moreButtonLink}>
          {<Label name="entry__continue_reading"/>} &raquo;
        </Link>}
      </p>
    </div>
  );
};
