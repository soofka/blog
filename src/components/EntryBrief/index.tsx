import * as React from 'react';
import { Link } from 'react-router-dom';

import Label from 'components/Label';
import AssetsProvider from 'common/AssetsProvider';

import {
  StyledEntryImageWrapper,
  StyledEntryImage,
  StyledEntryBriefContainer,
} from './styled';

export interface EntryBriefInterface {
  text: string;
  imageFileName?: string;
}

interface EntryBriefPropsInterface extends EntryBriefInterface {
  moreButtonLink?: string;
}

export const EntryBrief = (props: EntryBriefPropsInterface): JSX.Element => {
  const { text, imageFileName, moreButtonLink } = props;
  const imageUrl = AssetsProvider.getEntryImageFilePath(imageFileName);

  return (
    <div>
      {imageFileName &&
        <StyledEntryImageWrapper>
          <StyledEntryImage
            src={imageUrl}
          />
        </StyledEntryImageWrapper>
      }
      <StyledEntryBriefContainer dangerouslySetInnerHTML={{ __html: text }}></StyledEntryBriefContainer>
      <p>
        {moreButtonLink && <Link to={moreButtonLink}>
          {<Label name="entry__continue_reading"/>} &raquo;
        </Link>}
      </p>
    </div>
  );
};

export default EntryBrief;
