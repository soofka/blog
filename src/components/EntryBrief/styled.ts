import styledComponents from 'styled-components';

export const StyledEntryImageWrapper = styledComponents.div`
  width: 100%;
  overflow: hidden;
  margin: ${props => `${props.theme.marginS} 0 ${props.theme.marginM} 0`};
`;

export const StyledEntryImage = styledComponents.img`
  display: block;
  width: 100%;
  margin-top: -25%;
  margin-bottom: -25%;
`;

export default StyledEntryImageWrapper;
