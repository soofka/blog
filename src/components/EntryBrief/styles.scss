@import 'components/Blog/variables.scss';

$entryBriefImageHeight: 24 * $baseDistance;

.entry-brief--image-container {
  float: left;
  width: 100%;
  height: $entryBriefImageHeight;
  margin: $marginS 0 $marginM 0;
  padding: 0;
  border: 2px solid $white;
  outline: 1px solid $lighterBlack;
  background-repeat: no-repeat;
  background-position: center center;
  background-attachment: scroll;
  background-size: cover;
}
