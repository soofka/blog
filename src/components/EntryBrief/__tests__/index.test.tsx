import * as React from 'react';
import { shallow } from 'enzyme';

import EntryBrief from 'components/EntryBrief';

describe('<EntryBrief/>', () => {

  it('renders correctly', () => {
    expect(shallow(<EntryBrief text="test"/>)).toMatchSnapshot();
  });

  it('renders correctly with optional props', () => {
    expect(shallow(
    <EntryBrief
      text="tesText"
      imageFileName="testImageFileName"
      moreButtonLink="testMoreButtonLink"
    />)).toMatchSnapshot();
  });

});
