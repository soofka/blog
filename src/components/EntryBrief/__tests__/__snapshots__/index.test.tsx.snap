// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntryBrief/> renders correctly 1`] = `
<div>
  <p>
    testText
     
  </p>
</div>
`;

exports[`<EntryBrief/> renders correctly with optional props 1`] = `
<div>
  <styled.div>
    <styled.img
      src="/images/testImageFileName"
    />
  </styled.div>
  <p>
    testText
     
    <Link
      replace={false}
      to="testMoreButtonLink"
    >
      <inject-_class-with-languageStore
        name="entry__continue_reading"
      />
       Â»
    </Link>
  </p>
</div>
`;
